lib/Conversion/TorchOnnxToTorch/ComMicrosoftDomain.cpp

```diff
@@ -63,4 +63,393 @@ void mlir::torch::onnx_c::populateComMicrosoftDomain(
+              /*other=*/cstSequenceLength, /*alpha=*/cstIntOne);
+          Torch::ValueTensorType initPosIdsType = Torch::ValueTensorType::get(
+              context, {sequenceLength},
+              IntegerType::get(context, 64, IntegerType::Signed));
+          Value initPosIds = rewriter.create<Torch::AtenArangeOp>(
+              loc, initPosIdsType, cstSequenceLength, cstInt64Dtype,
+              /*layout=*/cstNone,
+              /*device=*/cstNone, /*pin_memory=*/cstNone);
+          Value repeatValuesList = rewriter.create<Torch::PrimListConstructOp>(
+              binder.getLoc(),
+              Torch::ListType::get(Torch::IntType::get(context)),
+              llvm::SmallVector<Value>{cstBatchSize, cstIntOne});
+          positionIdsB = rewriter.create<Torch::AtenRepeatOp>(
+              loc, positionIdsType, initPosIds, /*repeats=*/repeatValuesList);
+
+          Value cstIntMinusOne = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getI64IntegerAttr(1));
+          Value viewSizeList = rewriter.create<Torch::PrimListConstructOp>(
+              binder.getLoc(),
+              Torch::ListType::get(Torch::IntType::get(context)),
+              llvm::SmallVector<Value>{cstIntMinusOne, cstIntOne});
+
+          Torch::ValueTensorType seqLensViewType = Torch::ValueTensorType::get(
+              context, llvm::SmallVector<int64_t>{batchSize, 1},
+              IntegerType::get(context, 64, IntegerType::Signed));
+          pastSeqLens = rewriter.create<Torch::AtenViewOp>(
+              loc, seqLensViewType, pastSeqLens, viewSizeList);
+
+          positionIdsB = rewriter.create<Torch::AtenAddTensorOp>(
+              loc, positionIdsType, positionIdsB, pastSeqLens,
+              /*alpha=*/cstIntOne);
+
+          totalSeqLens = rewriter.create<Torch::AtenViewOp>(
+              loc, seqLensViewType, totalSeqLens, viewSizeList);
+          Value cond = rewriter.create<Torch::AtenLtTensorOp>(
+              loc,
+              positionIdsType.getWithSizesAndDtype(positionIdsType.getSizes(),
+                                                   rewriter.getI1Type()),
+              positionIdsB, totalSeqLens);
+
+          Value cstOneTensorDataList =
+              rewriter.create<Torch::PrimListConstructOp>(
+                  loc,
+                  rewriter.getType<Torch::ListType>(
+                      rewriter.getType<Torch::IntType>()),
+                  SmallVector<Value>{cstIntOne});
+          Value cstOneTensor = rewriter.create<Torch::AtenTensorOp>(
+              loc,
+              Torch::ValueTensorType::get(
+                  context, {},
+                  IntegerType::get(context, 64, IntegerType::Signed)),
+              cstOneTensorDataList, /*dtype=*/cstInt64Dtype,
+              /*layout=*/cstNone, /*requires_grad=*/cstFalse);
+
+          positionIdsB = rewriter.create<Torch::AtenWhereSelfOp>(
+              loc, positionIdsType, cond, positionIdsB, cstOneTensor);
+
```

Comment:
To align with onnx implementation. Better to add comments
// Initialize separate buffers for rotary embeddings
// Assume packed_qkv == false, qkv is always unpacked.

Reply:
Actually, this comment is not required since we are not creating any buffers here. The onnx implementation works slightly differently and is actually creating new tensors hence that comment is present there.

Reply:
Btw, I have added a TODO for the packed_qkv.